lib/ansible/modules/monitoring/zabbix/zabbix_user.py

```diff
@@ -0,0 +1,483 @@
+            state=dict(type='str', default="present", choices=['present', 'absent']),
+            timeout=dict(type='int', default=10)
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_ZABBIX_API:
+        module.fail_json(msg="Missing required zabbix-api module (check docs or install with: pip install zabbix-api)")
+
+    server_url = module.params['server_url']
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    http_login_user = module.params['http_login_user']
+    http_login_password = module.params['http_login_password']
+    validate_certs = module.params['validate_certs']
+    alias = module.params['alias']
+    name = module.params['name']
+    surname = module.params['surname']
+    usrgrps = module.params['usrgrps']
+    passwd = module.params['passwd']
+    lang = module.params['lang']
+    theme = module.params['theme']
+    autologin = module.params['autologin']
+    autologout = module.params['autologout']
+    refresh = module.params['refresh']
+    rows_per_page = module.params['rows_per_page']
+    after_login_url = module.params['after_login_url']
+    user_medias = module.params['user_medias']
+    user_type = module.params['type']
+    state = module.params['state']
+    timeout = module.params['timeout']
+
+    if autologin:
+        autologin = '1'
+    else:
+        autologin = '0'
+
+    user_type_dict = {
+        'Zabbix user': '1',
+        'Zabbix admin': '2',
+        'Zabbix super admin': '3'
+    }
+    user_type = user_type_dict[user_type]
+
+    zbx = None
+
+    # login to zabbix
+    try:
+        zbx = ZabbixAPI(server_url, timeout=timeout, user=http_login_user, passwd=http_login_password,
+                        validate_certs=validate_certs)
+        zbx.login(login_user, login_password)
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Zabbix server: %s" % e)
+
+    user = User(module, zbx)
+
+    user_ids = {}
+    zbx_user = user.check_user_exist(alias)
+    if state == 'present':
+        user_group_ids = user.get_usergroupid_by_user_group_name(usrgrps)
+        if zbx_user:
+            user_ids = user.update_user(zbx_user[0]['userid'], alias, name, surname, user_group_ids, passwd, lang,
+                                        theme, autologin, autologout, refresh, rows_per_page, after_login_url,
+                                        user_medias, user_type)
+        else:
+            user_ids = user.add_user(alias, name, surname, user_group_ids, passwd, lang, theme, autologin, autologout,
+                                     refresh, rows_per_page, after_login_url, user_medias, user_type)
+
+    if state == 'absent':
+        if zbx_user:
+            user_ids = user.delete_user(zbx_user[0]['userid'], alias)
+        else:
+            module.exit_json(changed=False)
+
+    module.exit_json(changed=True, user_ids=user_ids)
```

Comment:
result is always changed when rerunning playbook with your first example (create of zabbix user). This breaks module idempotency. I believe this can be solved by comparing all `update_user()` arguments with information returned from `user.check_user_exist(alias)`

Reply:
It's a nice idea!
I will try to that advice.

Reply:
Added `user_parameter_difference_check` method to compare user parameter.